docker/sagemaker/Dockerfile.endpoint

```diff
@@ -0,0 +1,75 @@
+# Docker file for building a docker image for running GraphStorm code on Amazon SageMaker
+# Note: Distributed graph partition will use another docker image which will come soon.
+
+ARG DEVICE=gpu
+ARG SOURCE
+
+FROM ${SOURCE:-763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.3.0-gpu-py311-cu121-ubuntu20.04-sagemaker} as branch-gpu
+ENV dev_type=GPU
+ARG TORCH_VERSION
+
+# Uninstall preset Pytorch packages
+RUN pip3 uninstall -y torchdata torch torchvision torchaudio
+
+# Install Dependencies to fit DGL GraphBolt requirements
+RUN pip3 install \
+        torchdata==0.9.0 \
+        pydantic==2.7.1 \
+        transformers==4.47.1 \
+        ogb==1.3.6 \
+    && rm -rf /root/.cache
+
+RUN pip3 install \
+    torch==2.3 torchvision torchaudio \
+    --index-url https://download.pytorch.org/whl/cu121 \
+    && rm -rf /root/.cache
+
+# Install DGL GPU version
+RUN pip3 install dgl==2.5.0+cu121 -f https://data.dgl.ai/wheels-internal/torch-2.3/cu121/repo.html \
+    && rm -rf /root/.cache
+
+FROM ${SOURCE:-763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.3.0-cpu-py311-ubuntu20.04-sagemaker} as branch-cpu
+ENV dev_type=CPU
+ARG TORCH_VERSION
+
+# Uninstall preset Pytorch packages
+RUN pip3 uninstall -y torchdata torch torchvision torchaudio
+
+# Install Dependencies to fit DGL GraphBolt requirements
+RUN pip3 install \
+        torchdata==0.9.0 \
+        pydantic==2.7.1 \
+        transformers==4.47.1 \
+        ogb==1.3.6 \
+    && rm -rf /root/.cache
+
+RUN pip3 install \
+    torch==2.3 torchvision torchaudio \
+    --index-url https://download.pytorch.org/whl/cpu \
+    && rm -rf /root/.cache
+
+# Install DGL CPU version
+RUN pip3 install dgl==2.5.0 -f https://data.dgl.ai/wheels-internal/torch-2.3/repo.html \
```

Comment:
Do we want DGL version to be customizable?

Reply:
I can make it an argument, but we should expect most of users will use dgl 2.5.0.